src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -1174,214 +1193,255 @@
+            if isinstance(original_prompt, list):
+                # Extract text from multimodal prompts
+                normalized_content = next((item["text"] for item in original_prompt if item.get("type") == "text"), str(original_prompt))
+            
+            # Prevent duplicate messages
+            if not (self.chat_history and 
+                    self.chat_history[-1].get("role") == "user" and 
+                    self.chat_history[-1].get("content") == normalized_content):
+                # Add user message to chat history BEFORE LLM call so handoffs can access it
+                self.chat_history.append({"role": "user", "content": normalized_content})
 
             final_response_text = None
```

Ruff (0.11.9)

Local variable `final_response_text` is assigned to but never used

Remove assignment to unused variable `final_response_text`

(F841)